qa/mc_sc_connected_nodes.py

```diff
@@ -80,63 +80,69 @@ def run_test(self):
         mainchain_block_height = self.sc_nodes_bootstrap_info.mainchain_block_height
         first_mainchain_node_block = first_mainchain_node.getblock(str(mainchain_block_height))
 
-        # check mainchain block inclusion for sidechain nodes 1 and 2
-        check_mainchan_block_inclusion(first_sidechain_node, sidechain_id, 1, 0,
-                                       first_mainchain_node_block,
-                                       [genesis_account[1]], [wallet_balance], True)
-        check_mainchan_block_inclusion(second_sidechain_node, sidechain_id, 1, 0,
-                                       first_mainchain_node_block,
-                                       [genesis_account[1]], [wallet_balance], True)
-
-        try:
-            # check mainchain block inclusion for sidechain node 3
-            check_mainchan_block_inclusion(third_sidechain_node, sidechain_id, 1, 0,
-                                           first_mainchain_node_block,
-                                           [genesis_account[1]],  [wallet_balance],True)
-            # SC node 2 should not to know information about blocks generated from MC node 0
-            # Otherwise the test must to fail
-            assert_equal(1, 2)
-            assert_equal(1, 2)
-        except AssertionError:
-            assert_equal(1, 1)
-
+        # verify genesis information for SC node 1, 2 and 3
+        # verify the mc block is included inside SC nodes 1, 2 and 3
+        sc_1_mc_block_inclusion = is_mainchain_block_included(first_sidechain_node, sidechain_id, 1, 0,
+                                                              first_mainchain_node_block)
+        sc_2_mc_block_inclusion = is_mainchain_block_included(second_sidechain_node, sidechain_id, 1, 0,
+                                                              first_mainchain_node_block)
+        sc_3_mc_block_inclusion = is_mainchain_block_included(third_sidechain_node, sidechain_id, 1, 0,
+                                                              first_mainchain_node_block)
+        assert_true(sc_1_mc_block_inclusion, "The mainchain block is not included for SC node 1.")
+        assert_true(sc_2_mc_block_inclusion, "The mainchain block is not included for SC node 2.")
+        assert_true(sc_3_mc_block_inclusion, "The mainchain block is not included for SC node 3.")
+
+        check_sidechain_boxes(first_sidechain_node, sidechain_id, [genesis_account.publicKey], 1, [wallet_balance])
```

Comment:
only first sc node should own the genesis balance. another nodes should be 0 balanced